src/main/java/org/brapi/test/BrAPITestServer/service/core/ListService.java

```diff
@@ -237,12 +234,16 @@ private void updateEntity(ListEntity entity, @Valid ListNewRequest list) throws
 		}
 
 		if (list.getData() != null) {
-			entity.setData(list.getData().stream().map((item) -> {
+			List<ListItemEntity> items = new ArrayList<>();
+			ListIterator<String> iter = list.getData().listIterator();
+			while (iter.hasNext()) {
 				ListItemEntity itemEntity = new ListItemEntity();
-				itemEntity.setItem(item);
+				itemEntity.setPosition(iter.nextIndex());
+				itemEntity.setItem(iter.next());
```

Comment:
It might be worth adding a null check here.
```suggestion
						if (list.getData() != null) {
			List<ListItemEntity> items = new ArrayList<>();
			ListIterator<String> iter = list.getData().listIterator();
			while (iter.hasNext()) {
				String item = iter.next();
				if (item != null) {  // Add null check here
					ListItemEntity itemEntity = new ListItemEntity();
					itemEntity.setPosition(iter.nextIndex());
					itemEntity.setItem(item);
					itemEntity.setList(entity);
					items.add(itemEntity);
				}
			}
			entity.setData(items);
		} else {
			entity.setData(new ArrayList<>());
		}
```

Reply:
Done.